src/txn/txn.c

```diff
@@ -2844,12 +2870,17 @@ __wt_verbose_dump_txn_one(
      * Log a message and return an error if error code and an optional error string has been passed.
      */
     if (0 != error_code) {
-        WT_RET_MSG(session, error_code, "%s, %s", buf, error_string != NULL ? error_string : "");
+        WT_ERR_MSG(session, error_code, "%s, %s", (char *)buf->data,
+          error_string != NULL ? error_string : "");
     } else {
-        WT_RET(__wt_msg(session, "%s", buf));
+        WT_ERR(__wt_msg(session, "%s", (char *)buf->data));
     }
 
-    return (0);
+err:
+    __wt_scr_free(session, &snapshot_buf);
+    __wt_scr_free(session, &buf);
```

Comment:
These two lines should be in alphabetical order.

Reply:
complete，thanks.